### PR TITLE
[BUGFIX] Spark Schema has unexpected field for `spark.sql.warehouse.dir`

### PR DESCRIPTION
**Pull request by Shinnnyshinshin**
- The `spark_config` property of `DatasourceConfigSchema` was dropping a field after being dumped into a dictionary because the `DatasourceConfigSchema` was loading the property as a `Dict`. 
- This PR proposes adjust the type of the `spark_config` property to `Raw`, which makes it consistent with [ExecutionEngine](https://github.com/great-expectations/great_expectations/blob/e51575b3c6bdf2dae384cc4483136054118e88eb/great_expectations/data_context/types/base.py#L779)
- Modifies a change in #3126

### Definition of Done
Please delete options that are not relevant.

- [x] My code follows the Great Expectations [style guide](https://docs.greatexpectations.io/docs/contributing/style_guides/code_style)
- [x] I have performed a [self-review](https://docs.greatexpectations.io/docs/contributing/contributing_checklist) of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added [unit tests](https://docs.greatexpectations.io/docs/contributing/contributing_test#writing-unit-and-integration-tests) where applicable and made sure that new and existing tests are passing.
- [x] I have run any local integration tests and made sure that nothing is broken.

Thank you for submitting!

### CODE REVIEW

**Comment by Shinnnyshinshin** [comment]
closing as this is being pursued by #5494

**Comment by Shinnnyshinshin** [comment]
re-opening to see if the error still comes up

File: .github/release_schedule.json [hunk 1]
```diff
@@ -6,5 +6,5 @@
     "2022-08-11": "0.15.18",
     "2022-08-18": "0.15.19",
     "2022-08-25": "0.15.20",
-    "2022-09-01": "0.15.21",
+    "2022-09-01": "0.15.21"
```
Comment (Shinnnyshinshin) [inline .github/release_schedule.json:9]: This is causing some failures in tests at the moment.

**Review by cdkini — APPROVED** [review]

File: .github/release_schedule.json (hunk 1 above)
Comment (cdkini) [reply .github/release_schedule.json:9]: oop sorry 😢

**Review by anthonyburdi — APPROVED** [review]

**Pull request merged**